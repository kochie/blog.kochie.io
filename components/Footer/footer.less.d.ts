declare const styles: {
  readonly container: string
  readonly link: string
  readonly links: string
  readonly divider: string
}
export = styles
