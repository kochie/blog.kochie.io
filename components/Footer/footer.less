.container {
    background-color: grey;
    width: 100vw;
    height: 100px;
}