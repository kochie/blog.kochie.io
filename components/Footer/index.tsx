import React from 'react'

import style from './footer.less'

export default () => {
    return (
        <div className={style.container}>

        </div>
    )
}