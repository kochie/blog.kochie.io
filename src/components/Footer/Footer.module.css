.copyright {
  --fa-secondary-color: black;
}

@media (--medium-viewport) and (orientation: landscape) {
  .container {
    /* padding: 5px 10% 50px 10%; */
    padding: 10px 25px;
    justify-content: space-between;
    align-items: inherit;
    flex-direction: row;
  }
}

@media (--large-viewport) and (orientation: landscape) {
  .container {
    max-width: 1080px;
  }
}

.links {
  display: flex;
  position: relative;
  margin: 0 0 16px 0;

}

@media (--medium-viewport) and (orientation: landscape) {
  .links {
    margin: auto 0;
  }
}

/* .link {
  position: relative;
  margin-left: 20px;
} */

.link > a {
  text-decoration: none;
  color: white;
}

.divider::before {
  content: '';
  position: absolute;
  top: 11px;
  left: -11px;
  display: block;
  width: 2px;
  height: 2px;
  background: #fff;
  border-radius: 100%;
}

.heading{
  position: relative;
  text-decoration: none;
  width: max-content;
  margin: 0;
  transition: ease-in-out 0.25s;
}

.heading:hover {
  color: white;
  cursor: pointer;
}

.heading:before {
  content: "";
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0px;
  left: 0;
  background-color: white;
  visibility: hidden;
  transform: scaleX(0);
  transition: ease-in-out 0.25s;
}

.heading:hover:before {
  visibility: visible;
  transform: scaleX(1);
}
