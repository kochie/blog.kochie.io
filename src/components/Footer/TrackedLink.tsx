'use client'

import { trackEvent } from 'fathom-client'
import Link from 'next/link'

export default function TrackedLink({
  href,
  name,
}: {
  href: string
  name: string
}) {
  return (
    <Link
      href={href}
      className="text-sm leading-6 text-gray-300 hover:text-white"
      onClick={() => trackEvent(name)}
    >
      {name}
    </Link>
  )
}
