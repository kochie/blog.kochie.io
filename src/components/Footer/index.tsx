import React from 'react'
import style from '../../styles/footer.module.scss'

interface FooterProps {
  title: string
  links: {
    name: string
    src: string
  }[]
}

export default ({ title, links }: FooterProps) => {
  return (
    <div className={style.container}>
      <p>
        {title} Â© {new Date().getFullYear()}
      </p>
      <div className={style.links}>
        {links.map((link, i: number) => {
          const classes = [style.link]
          i === 0 ? null : classes.push(style.divider)
          return (
            <div key={link.src} className={classes.join(' ')}>
              <a className={style.heading} href={link.src}>{link.name}</a>
            </div>
          )
        })}
      </div>
    </div>
  )
}
