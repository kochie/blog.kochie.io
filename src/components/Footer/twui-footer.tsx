import {
  faBluesky,
  faGithub,
  faMastodon,
  faTwitter,
} from '@fortawesome/free-brands-svg-icons'
import { faCopyright } from '@fortawesome/pro-duotone-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Image from 'next/image'
import Link from 'next/link'
import Logo from './blog-logo.svg'
import { trackEvent } from 'fathom-client'

const navigation = {
  affiliates: [{ name: 'Blogroll', href: 'https://blogroll.org/' }],
  friends: [
    { name: 'Hugo', href: 'https://hugo.md' },
    { name: 'Terence', href: 'https://terencehuynh.com/' },
    { name: 'Nicholas', href: 'https://nicholas.cloud/' },
    { name: 'Eric', href: 'https://ericjiang.dev/' },
    { name: 'Daniel', href: 'https://daniel.st/' },
    { name: 'Matt', href: 'https://mattseymour.substack.com/' },
  ],
  links: [
    { name: 'Me', href: 'https://me.kochie.io' },
    { name: 'Linkedin', href: 'https://linkedin.com/in/rkkochie' },
    {
      name: 'RSS',
      href: `https://${
        process.env.NEXT_PUBLIC_PROD_URL || process.env.NEXT_PUBLIC_VERCEL_URL
      }/feed/rss.xml`,
    },
  ],
  social: [
    {
      name: 'Bluesky',
      href: 'https://bsky.app/profile/kochie.bsky.social',
      icon: () => <FontAwesomeIcon icon={faBluesky} size="xl" className="" />,
    },
    {
      name: 'Twitter',
      href: 'https://twitter.com/kochie',
      icon: () => <FontAwesomeIcon icon={faTwitter} size="xl" className="" />,
    },
    {
      name: 'Github',
      href: 'https://github.com/kochie',
      icon: () => <FontAwesomeIcon icon={faGithub} size="xl" className="" />,
    },
    {
      name: 'Mastodon',
      href: 'https://melb.social/@kochie',
      icon: () => <FontAwesomeIcon icon={faMastodon} size="xl" className="" />,
    },
  ],
}

interface FooterProps {
  title: string
  description: string
}

export function Footer({ title, description }: FooterProps) {
  return (
    <footer className="bg-[#222222]" aria-labelledby="footer-heading">
      <h2 id="footer-heading" className="sr-only">
        Footer
      </h2>
      <div className="mx-auto lg:max-w-5xl pb-8">
        <div className="mt-16 border-t border-white/10 pt-8 sm:mt-20 lg:mt-24" />
        <div className="xl:grid xl:grid-cols-3 xl:gap-8">
          <div className="space-y-8">
            <Image className="" src={Logo} alt="Company name" />
            <p className="text-sm leading-6 text-gray-300">{description}</p>
            <div className="flex space-x-6">
              {navigation.social.map((item) => (
                <a
                  key={item.name}
                  href={item.href}
                  className="text-gray-500 hover:text-gray-400"
                >
                  <span className="sr-only">{item.name}</span>
                  <item.icon aria-hidden="true" />
                </a>
              ))}
            </div>
          </div>
          <div className="mt-16 grid grid-cols-2 gap-8 xl:col-span-2 xl:mt-0">
            <div className="md:grid md:grid-cols-2 md:gap-8">
              <div className="mt-10 md:mt-0 md:col-start-2">
                <h3 className="text-sm font-semibold leading-6 text-white">
                  Affiliates
                </h3>
                <ul role="list" className="mt-6 space-y-3">
                  {navigation.affiliates.map((item) => (
                    <li key={item.name}>
                      <Link
                        href={item.href}
                        className="text-sm leading-6 text-gray-300 hover:text-white"
                        onClick={() => trackEvent(item.name)}
                      >
                        {item.name}
                      </Link>
                    </li>
                  ))}
                </ul>
              </div>
            </div>
            <div className="md:grid md:grid-cols-2 md:gap-8">
              <div>
                <h3 className="text-sm font-semibold leading-6 text-white">
                  Friends
                </h3>
                <ul role="list" className="mt-6 space-y-3">
                  {navigation.friends.map((item) => (
                    <li key={item.name}>
                      <Link
                        href={item.href}
                        className="text-sm leading-6 text-gray-300 hover:text-white"
                        onClick={() => trackEvent(item.name)}
                      >
                        {item.name}
                      </Link>
                    </li>
                  ))}
                </ul>
              </div>
              <div className="mt-10 md:mt-0">
                <h3 className="text-sm font-semibold leading-6 text-white">
                  Links
                </h3>
                <ul role="list" className="mt-6 space-y-3">
                  {navigation.links.map((item) => (
                    <li key={item.name}>
                      <Link
                        href={item.href}
                        className="text-sm leading-6 text-gray-300 hover:text-white"
                        onClick={() => trackEvent(item.name)}
                      >
                        {item.name}
                      </Link>
                    </li>
                  ))}
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div className="mt-16 border-t border-white/10 pt-8 sm:mt-20 lg:mt-24">
          <p className="text-xs leading-5 text-gray-400">
            <span>{title}</span>{' '}
            <FontAwesomeIcon icon={faCopyright} className={''} />{' '}
            <span>2020</span>
          </p>
        </div>
      </div>
    </footer>
  )
}
