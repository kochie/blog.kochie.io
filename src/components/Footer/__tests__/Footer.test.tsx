import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faCopyright } from '@fortawesome/pro-duotone-svg-icons'

import Footer from '@/components/Footer'

beforeAll(() => {
  library.add(faCopyright)
})

describe('FOOTER COMPONENT', () => {
  test('renders correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(
        <Footer
          title={'testTitle'}
          links={[{ src: 'link', name: 'testLink', goal: 'goal' }]}
        />
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
