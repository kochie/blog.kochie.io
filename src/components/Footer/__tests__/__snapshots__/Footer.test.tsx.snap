// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FOOTER COMPONENT renders correctly 1`] = `
<div
  className="w-full bg-gray-500"
>
  <div
    className="flex flex-col justify-evenly md:content-evenly items-center px-4 text-white h-20 md:m-auto md:flex-row md:justify-between md:align-baseline lg:max-w-5xl"
  >
    <p>
      testTitle
       
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-copyright copyright"
        data-icon="copyright"
        data-prefix="fad"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-duotone-group"
          style={{}}
        >
          <path
            className="fa-secondary"
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM199.4 312.6c31.2 31.2 81.9 31.2 113.1 0c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9c-50 50-131 50-181 0s-50-131 0-181s131-50 181 0c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0c-31.2-31.2-81.9-31.2-113.1 0s-31.2 81.9 0 113.1z"
            fill="currentColor"
            style={{}}
          />
          <path
            className="fa-primary"
            d="M312.6 199.4c-31.2-31.2-81.9-31.2-113.1 0s-31.2 81.9 0 113.1s81.9 31.2 113.1 0c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9c-50 50-131 50-181 0s-50-131 0-181s131-50 181 0c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0z"
            fill="currentColor"
            style={{}}
          />
        </g>
      </svg>
       
      2020
    </p>
    <div
      className="flex relative"
    >
      <div
        className="relative ml-4"
      >
        <a
          className="heading"
          href="link"
          onClick={[Function]}
        >
          testLink
        </a>
      </div>
    </div>
  </div>
</div>
`;
